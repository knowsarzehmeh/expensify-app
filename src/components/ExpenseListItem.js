import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({
  dispatch,
  id,
  description,
  note,
  amount,
  createdAt
}) => {
  return (
    <div>
      <ul>
        <li>
          <h4>
            <Link to={`/edit/${id}`}>{description} </Link>
          </h4>
          {amount} - {createdAt}{" "}
        </li>
      </ul>
    </div>
  );
};

export default ExpenseListItem;
