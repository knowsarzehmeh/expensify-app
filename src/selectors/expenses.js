import moment from "moment";
// Get visible Expenses
const getVisibleExpenses = (expenses, { text, sortBy, startDate, endDate }) => {
  // expenses is an array, lets filter tru it
  return expenses
    .filter(expense => {
      const createdAtMoment = moment(expense.createdAt);
      const textMatch = expense.description
        .toLowerCase()
        .includes(text.toLowerCase());
      const startDateMatch = startDate
        ? startDate.isSameOrBefore(createdAtMoment, "day")
        : true;
      const endDateMatch = endDate
        ? endDate.isSameOrAfter(createdAtMoment, "day")
        : true;
      // const startDateMatch = typeof startDate !== 'number' || expense.createdAt >= startDate ;
      // const endDateMatch = typeof endDate !== 'number' || expense.createdAt <= endDate;

      return textMatch && startDateMatch && endDateMatch;
    })
    .sort((a, b) => {
      if (sortBy === "date") {
        return a.createdAt < b.createdAt ? 1 : -1;
      } else if (sortBy === "amount") {
        return a.amount < b.amount ? 1 : -1;
      } else {
        return 0;
      } // equal date or amount
    });
};

export default getVisibleExpenses;
