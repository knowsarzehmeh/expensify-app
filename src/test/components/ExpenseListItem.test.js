import React from "react";
import ExpenseListItem from "../../components/ExpenseListItem";
import { shallow } from "enzyme";
import expenses from "../redux/fixtures/expenses";

test("should render expense list item component with a single expense", () => {
  const wrapper = shallow(
    <ExpenseListItem key={expenses[0].id} {...expenses[0]} />
  );
  expect(wrapper).toMatchSnapshot();
});
