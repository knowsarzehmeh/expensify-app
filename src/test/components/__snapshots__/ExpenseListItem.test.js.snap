// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item component with a single expense 1`] = `
<div>
  <ul>
    <li>
      <h4>
        <Link
          replace={false}
          to="/edit/1"
        >
          Gum
           
        </Link>
      </h4>
      195
       - 
      0
       
    </li>
  </ul>
</div>
`;
